tests/nested/manual/muinstaller-core/task.yaml

```diff
@@ -0,0 +1,212 @@
+summary: End-to-end test for install Ubuntu Core via muinstaller
+
+systems: [ubuntu-22.04-64]
+
+environment:
+  # Test both encrypted and unencrypted install using the muinstaller
+  NESTED_ENABLE_TPM/encrypted: true
+  NESTED_ENABLE_SECURE_BOOT/encrypted: true
+
+  # unencrypted case
+  NESTED_ENABLE_TPM/plain: false
+  NESTED_ENABLE_SECURE_BOOT/plain: false
+
+  # ensure we use our latest code
+  NESTED_BUILD_SNAPD_FROM_CURRENT: true
+  NESTED_REPACK_KERNEL_SNAP: true
+  NESTED_ENABLE_OVMF: true
+  # image
+  IMAGE_MOUNTPOINT: /mnt/cloudimg
+
+prepare: |
+  if [ "$TRUST_TEST_KEYS" = "false" ]; then
+      echo "This test needs test keys to be trusted"
+      exit
+  fi
+  apt install dosfstools
+  snap install jq
+  snap install yq
+
+restore: |
+  rm -rf ./classic-root
+
+execute: |
+  # shellcheck source=tests/lib/prepare.sh
+  . "$TESTSLIB/prepare.sh"
+  #shellcheck source=tests/lib/nested.sh
+  . "$TESTSLIB"/nested.sh
+
+  version="$(nested_get_version)"
+
+  # Retrieve the gadget
+  snap download --basename=pc --channel="$version/edge" pc
+
+  # Modify gadget, making sure we can access the device (we are not building
+  # the image in the usual way in snapd spread tests).
+  unsquashfs -d pc-gadget pc.snap
+  echo 'console=ttyS0' > pc-gadget/cmdline.extra
+  cat <<EOF > pc-gadget/cloud.conf
+  #cloud-config
+  datasource_list: [NoCloud]
+  ssh_pwauth: True
+  users:
+   - name: user1
+     sudo: ALL=(ALL) NOPASSWD:ALL
+     shell: /bin/bash
+  chpasswd:
+    list: |
+      user1:ubuntu
+    expire: False
+  EOF
+
+  echo "Sign the shim binary"
+  KEY_NAME=$(tests.nested download snakeoil-key)
+  SNAKEOIL_KEY="$PWD/$KEY_NAME.key"
+  SNAKEOIL_CERT="$PWD/$KEY_NAME.pem"
+  tests.nested secboot-sign gadget pc-gadget "$SNAKEOIL_KEY" "$SNAKEOIL_CERT"
+  snap pack --filename=pc.snap pc-gadget/
+
+  # Retrieve kernel
+  snap download --basename=pc-kernel --channel="$version/edge" pc-kernel
+  # Build kernel with initramfs with the compiled snap-bootstrap
+  uc20_build_initramfs_kernel_snap "$PWD/pc-kernel.snap" "$NESTED_ASSETS_DIR"
+  mv "${NESTED_ASSETS_DIR}"/pc-kernel_*.snap pc-kernel.snap
+
+  # prepare a core seed
+  # TODO:
+  # - repacked snapd snap
```

Comment:
Something is inconsistent with this comment and the next line, maybe the next line should read `--snap=./local-snapd.snap` ?

Reply:
Changed